tests/functions/test_expand_grid.py

```diff
@@ -23,15 +23,76 @@ def test_not_a_dict_1():
         expand_grid(others=data)
 
 
-def test_empty_dict():
+empty_containers = [
+    {},
+    {"x": pd.Series([], dtype="int")},
+    {"x": pd.DataFrame([])},
+    {"x": [], "y": [2, 3]},
+    {"x": [2, 3], "y": set()},
+    {"x": np.array([])},
+]
+
+
+@pytest.mark.parametrize("empty", empty_containers)
+def test_empty(empty):
     """Test that entry should not be empty."""
-    data = {}
     with pytest.raises(ValueError):
-        expand_grid(others=data)
+        expand_grid(others=empty)
+
+
+frames_series = [
+    {"x": pd.Series([2, 3])},
+    {"x": pd.Series([2, 3], name="y")},
+    {"x": pd.DataFrame([[2, 3], [6, 7]])},
+]
+
+frames_series_output = [
+    pd.DataFrame([2, 3], columns=["x"]),
+    pd.DataFrame([2, 3], columns=["x_y"]),
+    pd.DataFrame([[2, 3], [6, 7]]).add_prefix("x_"),
+]
+
+zip_frame_series = zip(frames_series, frames_series_output)
+
+
+@pytest.mark.parametrize("frames_series,outputs", zip_frame_series)
+def test_frames_series_single_index(frames_series, outputs):
+    """Test for single indexed dataframe and series."""
+    assert_frame_equal(expand_grid(others=frames_series), outputs)
+
+
+multiIndex_pandas = [
+    {
+        "x": pd.Series(
+            [2, 3, 4],
+            name="mika",
+            index=pd.MultiIndex.from_tuples([(1, 2), (3, 4), (5, 6)]),
+        )
+    },
+    {
+        "x": pd.DataFrame(
+            [[2, 3, 4]],
+            columns=pd.MultiIndex.from_tuples([(1, 2), (3, 4), (5, 6)]),
+        )
+    },
+]
+
+multiIndex_output = [
+    pd.DataFrame({"x_mika": [2, 3, 4]}),
+    pd.DataFrame({"x_0": [2], "x_1": [3], "x_2": [4]}),
+]
+
+zip_multiIndex = zip(multiIndex_pandas, multiIndex_output)
+
+
+@pytest.mark.parametrize("multiIndex_data,multiIndex_outputs", zip_multiIndex)
```

Comment:
@UGuntupalli in case it helps, this is an example of how parametrization in testing is done! (No action needed here from you, just wanted to get your attention on this.)